### PR TITLE
chore: refactor token utils & impl a new service

### PR DESCRIPTION
**Pull request by yasinmiran**

### CODE REVIEW

File: services/localega-tsd-proxy/src/main/java/no/elixir/fega/ltp/services/TokenService.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+   * @param passportScopedAccessToken the Passport-Scoped Access Token (a JWS-encoded JWT) used to
+   *     authenticate and retrieve Visa tokens. It must be valid and include the `ga4gh_passport_v1`
+   *     scope.
+   * @return a {@link List} of {@link Visa} objects representing the verified Visa tokens.
+   * @throws IllegalArgumentException if the access token is invalid, cannot be verified, or if
+   *     required public key files are missing or unreadable.
+   * @link <a href="https://ga4gh.github.io/data-security/aai-openid-connect-profile">docs</a>
+   * @link <a href="https://ga4gh-echo.aai.lifescience-ri.eu/index.html">playground</a>
+   */
+  public List<Visa> fetchTheFullPassportUsingPassportScopedAccessTokenAndGetVisas(
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply services/localega-tsd-proxy/src/main/java/no/elixir/fega/ltp/services/TokenService.java:151]: Ignoring this suggestion as this method is used by Export Request feature.

Reasoning: as this is a service, the caller should be able to access this method as well. The `getControlledAccessGrantsVisas`is something handy but its something we should avoid for Export Request feature.

File: services/localega-tsd-proxy/src/main/java/no/elixir/fega/ltp/services/TokenService.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+   * multiple issuers.
+   *
+   * @param visaToken the Visa token (a JWS-encoded JWT) to be verified and transformed.
+   * @return an {@link Optional} containing the {@link Visa} object if the token is valid;
+   *     otherwise, an empty {@link Optional}.
+   * @throws IllegalArgumentException if the Visa token cannot be verified due to issues such as
+   *     invalid token format, missing claims, or failure to access required resources (e.g., public
+   *     key file or JWK endpoint).
+   */
+  public Optional<Visa> verifyVisaTokenAndTransformToVisaObject(String visaToken) {
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply services/localega-tsd-proxy/src/main/java/no/elixir/fega/ltp/services/TokenService.java]: Addressed in commit: [ad5b2e8](https://github.com/ELIXIR-NO/FEGA-Norway/pull/302/commits/ad5b2e8c0dba10e878dc457e39195c2141dfafcd)

File: services/localega-tsd-proxy/src/main/java/no/elixir/fega/ltp/services/TokenService.java [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+   *
+   * @param jwtToken the JWT token to extract a fragment from; must be a valid JWS-encoded JWT.
+   * @param tokenFragment the fragment to extract, specified using the {@link TokenFragment} enum.
+   *     The enum values correspond to the ordinal positions of the fragments in the JWT (0 for
+   *     header, 1 for payload, 2 for signature).
+   * @return a {@link JsonObject} representing the decoded fragment.
+   * @throws IllegalArgumentException if the JWT token format is invalid or the specified fragment
+   *     is missing.
+   */
+  public JsonObject extractFragmentFromJWT(String jwtToken, TokenFragment tokenFragment) {
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply services/localega-tsd-proxy/src/main/java/no/elixir/fega/ltp/services/TokenService.java]: Addressed in commit: [ad5b2e8](https://github.com/ELIXIR-NO/FEGA-Norway/pull/302/commits/ad5b2e8c0dba10e878dc457e39195c2141dfafcd)

**Review by kjellp — APPROVED** [review]
LGTM

**Pull request merged**